dynamodb/README.md

```diff
@@ -0,0 +1,134 @@
+# HarbourBridge: Turnkey DynamoDB-to-Spanner Evaluation
+
+HarbourBridge is a stand-alone open source tool for Cloud Spanner evaluation,
+using data from an existing PostgreSQL or MySQL or DynamoDB database. This
+README provides details of the tool's DynamoDB capabilities. For general
+HarbourBridge information see this [README](https://github.com/cloudspannerecosystem/harbourbridge#harbourbridge-turnkey-spanner-evaluation).
+
+## Example DynamoDB Usage
+
+The following examples assume `harbourbridge` has been added to your PATH
+environment variable.
+
+HarbourBridge can be run directly on a DynamoDB database (via the aws go sdk
+package).
+
+Before running HarbourBridge, make sure that you have
+[set up your AWS credentials/region](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html)
+correctly.
+
+To use the tool directly on a DynamoDB database (it will migrate all tables),
+run
+
+```sh
+harbourbridge -driver=dynamodb
+```
+
+Due the schemaless nature of DynamoDB, the tool infers the schema based on a
+certain amount of sampled data, by default, 100000 rows. If a table has more
+rows than the default value, we only use 100000 rows for estimating the schema.
+You can change this value, run
+
+```sh
+harbourbridge -driver=dynamodb -schema-sample-size=1000000
+```
+
+## Schema Conversion
+
+The HarbourBridge tool maps DynamoDB types to Spanner types as follows:
+
+| PostgreSQL Type    | Spanner Type               | Notes                                     |
+| ------------------ | -------------------------- | ----------------------------------------- |
+| `Number`           | `NUMERIC` or `STRING`      | defaults to NUMERIC, otherwise, STRING    |
+| `String`           | `STRING`                   |                                           |
+| `Boolean`          | `BOOL`                     |                                           |
+| `Binary`           | `BYTES`                    |                                           |
+| `Null`             | A nullable column type     |                                           |
+| `List`             | `STRING`                   | json encoding                             |
+| `Map`              | `STRING`                   | json encoding                             |
+| `StringSet`        | `ARRAY<STRING>`            |                                           |
+| `NumberSet`        | `ARRAY<NUMERIC or STRING>` |                                           |
+| `BinarySet`        | `ARRAY<BYTES>`             |                                           |
+
+We discuss these, as well as other limits and notes on schema conversion, in the
+following sections.
+
+### Schema Inference
+
+DynamoDB is schemaless and it is impossible to get the full information of the
```

Comment:
Suggest rewording:

DynamoDB is schemaless database: other than a primary index and optional
secondary index, column names and types are essentially unconstrained
and can vary from one row to the next.

However, many customers use DynamoDB in a consistent, structured way
with a fairly well defined set of columns and types. Our HarbourBridge support
for DynamoDB focuses on this use-case, and we construct a Spanner schema
by inspecting table data.

For small tables, we inspect all rows of the table. For large tables, scanning the
entire table would be extremely expensive and slow, and so we only inspect the
first N rows (defined by the flag `schema-sample-size`) from the table scan.
While DynamoDB doesn't return scan results in order, they might not be a truly
random sample of rows. However, the alternative of randomly sampling rows
would be much more expensive. 

Columns with consistent types are assigned Spanner types as detailed below.
Columns without a consistent type are mapped to STRING.